src/Microsoft.DotNet.Build.Tasks/GenerateEncodingTable.cs

```diff
@@ -0,0 +1,395 @@
+﻿using Microsoft.Build.Framework;
```

Comment:
if you'll auto generate the file during the build time then it is not a good idea to generate the file in the source directory because this will have github show there is a new change in the sources. it'll be better to generate the file in the intermediate folder and reference it from there. 
to include this file in the project you have to do it in a target element. you may look at the file resources.targets in the build tools in the line 
      <Compile Include="$(IntermediateResOutputFileFullPath)" />